pkg/container/kubernetes_runner.go

```
@@ -0,0 +1,518 @@
+	return false
+}
+
+// KubernetesRunnerConfig handles the configuration for the Kubernetes runner
+// It sources from various locations, in the following order of precedence:
+//  1. The "global" config file
+//  2. Defaults
+//
+// TODO: Add loaders from package config and environment
+type KubernetesRunnerConfig struct {
+	Provider    string            `json:"provider" yaml:"provider"`
+	Repo        string            `json:"repo" yaml:"repo"`
+	Namespace   string            `json:"namespace" yaml:"namespace"`
+	Annotations map[string]string `json:"annotations,omitempty" yaml:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"`
+
+	StartTimeout time.Duration `json:"startTimeout" yaml:"startTimeout"`
+	BuildTimeout time.Duration `json:"buildTimeout" yaml:"buildTimeout"`
+
+	PodTemplate *KubernetesRunnerConfigPodTemplate `json:"podTemplate,omitempty" yaml:"podTemplate,omitempty" ignored:"true"`
+
+	// A "burstable" QOS is really the only thing that makes sense for ephemeral builder pods
+	// Only set the requests, and not the limits
+	Resources corev1.ResourceList
+
+	baseConfigFile string
+}
+
+type KubernetesRunnerConfigPodTemplate struct {
+	NodeSelector     map[string]string `json:"nodeSelector,omitempty" yaml:"nodeSelector,omitempty"`
+	Env              []corev1.EnvVar   `json:"env,omitempty" yaml:"env,omitempty"`
+	Affinity         *corev1.Affinity  `json:"affinity,omitempty" yaml:"affinity,omitempty"`
+	RuntimeClassName *string           `json:"runtimeClassName,omitempty" yaml:"runtimeClassName,omitempty"`
+	Volumes          []corev1.Volume   `json:"volumes,omitempty" yaml:"volumes,omitempty"`
+}
+
+// NewKubernetesConfig returns a default Kubernetes runner config setup
+func NewKubernetesConfig(opt ...KubernetesRunnerConfigOptions) *KubernetesRunnerConfig {
+	cfg := &KubernetesRunnerConfig{
+		Provider:     "generic",
+		Namespace:    "default",
+		Repo:         "ttl.sh/melange",
```

This pushes to `ttl.sh` by default? I don't think we should do that.

---

+1, but I think we can fix this in a followup PR, where the repo must be configured somehow. This is sort of like requiring `KO_DOCKER_REPO` to be set basically.

---

I've got it staged as a follow up, right now it's overlayed as `defaults -> .melange.k8s.yaml -> package config -> env var`, but this can be changed.

I'd like to keep the config handling as a separate PR though, as it somewhat detracts from the basic functionality of this.

However if we're worried about silently defaulting to pushing to a public repo (a valid concern), I'm ok with erroring out when no repo is specified for now!